addons/mail/static/src/models/mobile_messaging_navbar/mobile_messaging_navbar.js

```diff
@@ -0,0 +1,82 @@
+/** @odoo-module **/
+
+import { registerModel } from '@mail/model/model_core';
+import { attr, one2one } from '@mail/model/model_field';
+import { clear, decrement, increment, insert, insertAndReplace, link, replace, unlink } from '@mail/model/model_field_command';
+
+registerModel({
+    name: 'mail.mobile_messaging_navbar',
+    identifyingFields: [['discuss', 'messagingMenu']],
+    recordMethods: {
+        /**
+         * @private
+         * @returns {string}
+         */
+        _computeActiveTabId() {
+            if (this.discuss) {
+                return this.discuss.activeMobileNavbarTabId
+            }
+            if (this.messagingMenu) {
+                return this.messagingMenu.activeTabId;
+            }
+            return clear();
+        },
+        /**
+         * @private
+         * @returns {Object[]}
+         */
+        _computeTabs() {
+            if (this.discuss) {
+                return [{
+                    icon: 'fa fa-inbox',
+                    id: 'mailbox',
+                    label: this.env._t("Mailboxes"),
+                }, {
+                    icon: 'fa fa-user',
+                    id: 'chat',
+                    label: this.env._t("Chat"),
+                }, {
+                    icon: 'fa fa-users',
+                    id: 'channel',
+                    label: this.env._t("Channel"),
+                }];
+            }
+            if (this.messagingMenu) {
+                return [{
+                    icon: 'fa fa-envelope',
+                    id: 'all',
+                    label: this.env._t("All"),
+                }, {
+                    icon: 'fa fa-user',
+                    id: 'chat',
+                    label: this.env._t("Chat"),
+                }, {
+                    icon: 'fa fa-users',
+                    id: 'channel',
+                    label: this.env._t("Channel"),
+                }];
+            }
+            return [];
+        },
+    },
+    fields: {
+        /**
+         * Tab selected in the messaging menu.
+         * Either 'all', 'chat' or 'channel'.
+         */
+        activeTabId: attr({
+            compute: '_computeActiveTabId',
+        }),
+        discuss: one2one('mail.discuss', {
+            inverse: 'mobileMessagingNavbar',
+            readonly: true,
+        }),
+        messagingMenu: one2one('mail.messaging_menu', {
+            inverse: 'mobileMessagingNavbar',
+            readonly: true,
+        }),
+        tabs: attr({
+            compute: '_computeTabs',
+        }),
```

Comment:
It would be better as a relation to new model instead of array, but acceptable for now. I would at least suggest to document what it contains then.